src/RA_Dlg_Login.cpp

```diff
@@ -46,9 +51,32 @@ INT_PTR CALLBACK RA_Dlg_Login::RA_Dlg_LoginProc(HWND hDlg, UINT uMsg, WPARAM wPa
                         return TRUE;
                     }
 
+
+                    //URL-escape password so it gets sent to the server properly
+                    //https://stackoverflow.com/questions/154536/encode-decode-urls-in-c
+                    std::ostringstream escaped;
+                    escaped.fill('0');
+                    escaped << std::hex;
+
+                    const std::string &value = ra::Narrow(sPassEntry);
+                    for (std::string::const_iterator i = value.begin(), n = value.end(); i != n; ++i) {
+                        std::string::value_type c = (*i);
+
+                        // Keep alphanumeric and other accepted characters intact
+                        if (isalnum(c) || c == '-' || c == '_' || c == '.' || c == '~') {
+                            escaped << c;
+                            continue;
+                        }
+
+                        // Any other characters are percent-encoded
+                        escaped << std::uppercase;
+                        escaped << '%' << std::setw(2) << int((unsigned char)c);
```

Comment:
I don't think boxing a type is required for `ostream `as it's insertion operator (`operator<<`) has overloads for all primitive types.